### PR TITLE
Feature/input component

### PR DESCRIPTION
**Pull request by hungkail**
## Intro
Implemented by input and pat-ui Icon component.

## Usage
```
<Input {props} />
```
Check ReadMe for more detail.

### Further Improvements
1. Labeled property.
2. Action property.

### CODE REVIEW

File: dist/index.css [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@import url("https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/components/icon.min.css");
```
Comment (YuhanLin1105) [inline dist/index.css:1]: we should not use this css for the icon, ask Vencent

File: dist/index.css [hunk 2]
```diff
@@ -384,3 +385,96 @@ main {
   .btn-link:focus, .btn-link.focus {
     text-decoration: underline;
     box-shadow: none; }
+
+.ui.input {
```
Comment (YuhanLin1105) [inline dist/index.css:389]: do not put your css in this file directly. check how I did for button. 
And do not use .ui.input selecotor

File: package.json [hunk 3]
```diff
@@ -20,7 +20,11 @@
   "main": "./dist/index.js",
   "module": "./dist/index.js",
   "types": "./dist/index.d.ts",
-  "dependencies": {},
+  "dependencies": {
+    "@types/storybook__addon-info": "^5.2.1",
```
Comment (YuhanLin1105) [inline package.json:24]: we do not need storyboook for the libarary user. put it to the devdependencies

File: package.json [hunk 4]
```diff
@@ -20,7 +20,11 @@
...
+    "semantic-ui-css": "^2.4.1",
+    "semantic-ui-react": "^1.1.1"
```
Comment (YuhanLin1105) [inline package.json:26]: remove semantic-ui and semantic-ui-react

File: src/components/Input/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import React, { InputHTMLAttributes, FC } from 'react';
+import { classNames } from '../../utils/classNames';
+import Icon from '../Icon/Icon';
+
+interface IInputProps {
+  /** set customized style */
+  className?: string;
+  /** set input bar size */
+  size?: 'mini' | 'small' | 'large' | 'big' | 'huge' | 'massive';
```
Comment (YuhanLin1105) [inline src/components/Input/Input.tsx:9]: try to not use size direactly, you could use inputSize

File: storybook-static/index.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
+    overflow: hidden;
```
Comment (YuhanLin1105) [inline storybook-static/index.html:2]: Remove the storybook-static folder

File: dist/index.css (hunk 1 above)
Comment (hungkail) [reply dist/index.css:1]: done, wasn't using it, just forget to rebuild the library after I use Vincent's Icon

File: package.json (hunk 3 above)
Comment (hungkail) [reply package.json:24]: done

File: package.json (hunk 4 above)
Comment (hungkail) [reply package.json:26]: done

File: src/components/Input/Input.tsx (hunk 5 above)
Comment (hungkail) [reply src/components/Input/Input.tsx:9]: done

File: storybook-static/index.html (hunk 6 above)
Comment (hungkail) [reply storybook-static/index.html:2]: done

File: src/components/Input/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    else if (loading) {
+      iconName = 'spinner';
+    }
+  }
+  // const inputIcon = <i aria-hidden={'true'} className={iconClasses}></i>;
+  const patIcon = <Icon aria-hidden={'true'} name={iconName} loading={!!loading} disabled={!!disabled} size={inputSize? inputSize: 'small'}></Icon>;
+  return (
+    <div className={styleClasses}>
+      <input type={'text'} disabled={disabled} {...rest} />
+      {icon || loading ? patIcon : null}
```
Comment (YuhanLin1105) [inline src/components/Input/Input.tsx:86]: icon looks too bid. 
Screen Shot 2020-08-10 at 11 38 38 AM

File: src/components/Input/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  icon?: string | { [key: string]: string | boolean };
+  /** apply loading style to input bar */
+  loading?: boolean;
+  /** set icon to show on left or right, default is right */
+  iconPosition?: 'left' | 'right';
+}
+
+export type PatInputProps = IInputProps & InputHTMLAttributes<HTMLInputElement>;
+
+const Input: FC<PatInputProps> = (props) => {
```
Comment (YuhanLin1105) [inline src/components/Input/Input.tsx:30]: Add a description about how to import it.

File: src/components/Input/Input.tsx (hunk 7 above)
Comment (hungkail) [reply src/components/Input/Input.tsx:93]: fixed

File: src/components/Input/Input.tsx (hunk 8 above)
Comment (hungkail) [reply src/components/Input/Input.tsx:37]: added

**Pull request merged**